src/app/api/health/checks.ts

```diff
@@ -0,0 +1,68 @@
+import { BlockChangeLog } from '@/app/api/utils/db.schema'
+import { config } from '@/config'
+import { STATE_SYNC_BLOCK_STALENESS_THRESHOLD } from '@/lib/constants'
+import { db } from '@/lib/db'
+import { getBlockNumber } from 'wagmi/actions'
+
+export class WrongHealthCheckTypeError extends Error {
+  constructor(type: string) {
+    super(`Health check type "${type}" is not defined`)
+    this.name = 'WrongHealthCheckTypeError'
+  }
+}
+
+export class BlockNumberFetchError extends Error {
+  constructor() {
+    super(`Failed to fetch block number`)
+    this.name = 'BlockNumberFetchError'
+  }
+}
+
+const HEALTH_CHECK_TYPE = ['lastBlockNumber'] as const
+
+export type HealthCheckType = (typeof HEALTH_CHECK_TYPE)[number] | 'all'
+
+export type HealthCheckResult = boolean
+
+export const lastBlockNumber = async (): Promise<boolean> => {
+  const blockChangeLog = await db<BlockChangeLog>('BlockChangeLog')
+    .select('BlockChangeLog')
+    .orderBy('timestamp', 'desc')
+    .first()
+
+  if (!blockChangeLog) {
+    return false
+  }
+
+  const blockNumberOnChain = await getBlockNumber(config)
+
+  if (!blockNumberOnChain) {
+    throw new BlockNumberFetchError()
+  }
+
+  return blockChangeLog.blockNumber + BigInt(STATE_SYNC_BLOCK_STALENESS_THRESHOLD) >= blockNumberOnChain
```

Comment:
```suggestion
  return BigInt(blockChangeLog.blockNumber) + BigInt(STATE_SYNC_BLOCK_STALENESS_THRESHOLD) >= blockNumberOnChain
```
Knex does not parse to bigint

Reply:
ah yea I forgot about it. well, to be precise, knex does, but the postgres lib doesn't. I'll add a parser to the config.